test/unit/BPool.t.sol

```diff
@@ -6,15 +6,31 @@ import {BPool} from 'contracts/BPool.sol';
 import {IERC20} from 'contracts/BToken.sol';
 import {Test} from 'forge-std/Test.sol';
 import {LibString} from 'solmate/utils/LibString.sol';
+import {Utils} from 'test/unit/Utils.sol';
 
-abstract contract Base is Test, BConst {
+// TODO: remove once `private` keyword is removed in all test cases
+/* solhint-disable */
+
+abstract contract BasePoolTest is Test, BConst, Utils {
   using LibString for *;
 
   uint256 public constant TOKENS_AMOUNT = 3;
 
+  struct FuzzScenario {
+    uint256 poolAmountOut;
+    uint256 initPoolSupply;
+    uint256[TOKENS_AMOUNT] balance;
+  }
```

Comment:
NE! this is special to the Join test, other ones will have other scenarios, as there are other relevant variables to account, plz move this and the happy path back to the Join scope 🙏